File: 1470-shuffle-the-array/1470-shuffle-the-array.cpp
class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        size_t middle = nums.size() / 2;
        vector<int> firstHalf(nums.begin(), nums.begin() + middle);
        vector<int> secondHalf(nums.begin() + middle, nums.end());
        vector<int> shuffle;
        for(int i = 0; i < middle; i++){
            shuffle.push_back(firstHalf[i]);
            shuffle.push_back(secondHalf[i]);
        }
        return shuffle;
    }
};

File: 1470-shuffle-the-array/1470-shuffle-the-array.py
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        nums1 = nums[0:n]
        nums2 = nums[n:2*n]

        shuffle_arr = []
        for i in range(n):
            shuffle_arr.append(nums1[i])
            shuffle_arr.append(nums2[i])
        
        return shuffle_arr